### PR TITLE
Lhe producer update -DYToll and WTolNu

### PR DESCRIPTION
**Pull request by joonblee**
The DYToll and WTolNu code cannot produce the proper samples. Thanks to Alexander, there are some modification on the LHE producer.

### CODE REVIEW

**Comment by perrozzi** [comment]
+1

**Comment by perrozzi** [comment]
please test

**Comment by davidlange6** [comment]
hi @perrozzi - remind me why this should be in the release ? (We don't seem to use it for relvals)

**Comment by joonblee** [comment]
Hi @davidlange6 ,
The relval cannot produce DYToll LO and WTolNu LO samples. There is 0 events. You can see it at below link.
https://dmytro.web.cern.ch/dmytro/cmsprodmon/requests.php?campaign=CMSSW_9_4_0__extendedgen-1510338015

**Comment by davidlange6** [comment]
hum - strange I don't find the workflow in the release then. what is its number in runTheMatrix?

**Comment by prebello** [comment]
Hi David, 522.0 and 513.0

 9_4_0_pre3 but 9_4_0 and 10_0_0_pre1 also with these issues

Links below:

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_9_4_0__extendedgen-1510338015-DYTollJets_LO_Mad_13TeV_py8_Ta
this is 522.0

and

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_9_4_0__extendedgen-1510338015-WTolNuJets_LO_Mad_13TeV_py8
this is 513.0

9_4_0_pre3 informed in the HN https://hypernews.cern.ch/HyperNews/CMS/get/generators/3552.html
https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_9_3_0_pre4__extendedgen-1503283640-DYTollJets_LO_Mad_13TeV_py8_Ta
https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_9_3_0_pre4__extendedgen-1503283640-WTolNuJets_LO_Mad_13TeV_py8

The 10_0_0_pre1 had the same issues in these wfs as well:

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_10_0_0_pre1__extendedgen-1511097960-DYTollJets_LO_Mad_13TeV_py8_Ta

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_10_0_0_pre1__extendedgen-1511097960-WTolNuJets_LO_Mad_13TeV_py8


Cheers Pat
________________________________

**Comment by davidlange6** [comment]
Right - i guess this has been broken for a long while (since 92x?). it looks like the proper fix is to update /cvmfs/cms.cern.ch/phys_generator/gridpacks/slc6_amd64_gcc481/13TeV/madgraph/V5_2.4.2/dyellell01234j_5f_LO_MLM/v1/dyellell01234j_5f_LO_MLM_tarball.tar.xz (and similarly for the other workflow)  for the current cms compiler/architecture..

**Comment by agrohsje** [comment]
@davidlange6 
as we only want to update the gridpacks when there is a new mg5_amcatnlo version and not just because we changed gcc or similar in cmssw, we changed the lhe producer such that it picks the original version of cmssw used for the gridpack to produce the lhe text file and converts this to edm in the cmssw version of the release. this is now the new standard and the cmssw version as well as the arch are given in the gridpack name. the above modification of the relval fragments still specify the settings by hand for the old gridpacks. once we finished the validation of 2.6. we will update the gen fragments and the new gridpack will pick the proper releases automatically as given in the name. 
in summary: the proposed modifications by joonbin are ok and in sync with the new strategy of the gen group. can this please be merged in the next release so that mg5_amcatnlo workflows work again?

**Comment by perrozzi** [comment]
just to stress that the GEN conveners support the modification, as alexander says the GEN strategy is to have gridpacks working across different architectures, cfr for instance https://github.com/cms-sw/cmssw/pull/20676

**Comment by prebello** [comment]
Dear all,
would you have updates about this PR?
Errors in GEN Relvals should be fixed by it.
Could it be merged in 10_0_0_pre3?
Thank you.

**Comment by agrohsje** [comment]
@davidlange6  @perrozzi 
When can this PR be merged? Would be good to get it back in relval!

**Comment by davidlange6** [comment]
hi @perrozzi - that is "ok" for production where there are lots of grid packs and where you'd like to use a well controlled setup - for relval instead, there are a few and the general desire to test new things that get integrated (eg, on request of the GEN group). 

so I don't support using 71x to build and test 100x relvals.

**Comment by perrozzi** [comment]
ciao @davidlange6, sorry but I disagree.
I believe that the spirit of the central Relvals is to test the CMSSW software with small production.
The testing of gridpacks and more generally matrix elements is done in a separate way, by producing either private or gen-only requests and comparing between different versions making use of the RelVals diff tools (plus custom ones), cfr https://indico.cern.ch/event/673265/sessions/254029/attachments/1573257/2483183/agrohsje_genvali.pdf
This was agreed time ago with PPD.

Thus, I believe that we should not change the CMSSW release used to produce the gridpacks for each CMSSW we want to test (you might also remember that some combinations of CMSSW and matrix element versions don't go well along).
If needed, we can discuss further today during the ORP

**Comment by davidlange6** [comment]
ok, so looking again these are for gen validations - so its a  gen-only relval testing 71x software in 100x? 

That is a test of relevance to 71x (eg, when 71x is updated). What components are we testing that are actually in the 100x software stack, aside from the mechanics of the forking process? Maybe we can make life much simpler here?

**Comment by agrohsje** [comment]
@davidlange6 
No, we are not just testing 71x. These workflows include matching/merging. So they also probe the interplay of Pythia and the ME generator. It is just that changing cmssw for lhe generation is meaningless for us as we don't do it in production (and would mean additional work).

**Comment by agrohsje** [comment]
@davidlange6 
No, we are not just testing 71x. These workflows include matching/merging. So they also probe the interplay of Pythia and the ME generator. It is just that changing cmssw for lhe generation is meaningless for us as we don't do it in production (and would mean additional work).

**Comment by davidlange6** [comment]
as for productions... one might eventually think of future productions, which why we have development releases... the current one is already underway after all. Anyway,  we are told all the gen physics validations happen elsewhere... so its really the mechanics that are being tested by these workflows, so the agreed plan is to keep a workflow with a simple to support process to ensure that all continues to technically work.

**Comment by perrozzi** [comment]
having clarified what to do for validation (i.e. technical validation through RelVals, physics validation through GenOnly), can this PR be merged as it is please?

**Comment by davidlange6** [comment]
ok, i had a different impression about the plan in mind - but ok, lets see how this goes

**Comment by davidlange6** [comment]
+1
